src/core_plugins/timelion/server/handlers/__tests__/parse_sheet.js

```diff
@@ -0,0 +1,31 @@
+const parseSheet = require('../lib/parse_sheet');
+
+const expect = require('chai').expect;
+
+describe('timelion parse_sheet function', function () {
+
+  it('does not allow whitespace used between two expressions', async function () {
+    const data = ['.es() .es(404)'];
+    const ast = parseSheet(data);
+
+    expect(ast[0].length).to.equal(1);
+    expect(ast[0][0].type).to.equal('chain');
+  });
+
+  it('allows comma-delimited expressions', function () {
+    const data = ['.es(), .es(404)'];
+    const ast = parseSheet(data);
+
+    expect(ast[0].length).to.equal(2);
+    expect(ast[0][0].type).to.equal('chain');
+    expect(ast[0][1].type).to.equal('chain');
+  });
+
+  it('allows new line for white space', function () {
+    const data = [`.es()\n\r ,\n\r .es(404)`];
+    const ast = parseSheet(data);
+    expect(ast[0].length).to.equal(2);
+    expect(ast[0][0].type).to.equal('chain');
+    expect(ast[0][1].type).to.equal('chain');
+  });
+});
```

Comment:
I was a little confused by the use of "allow" and "does not allow" in the test descriptions, because I assumed we'd be throwing errors or somehow preventing disallowed input. After I dug into the tests, I realized this was an incorrect assumption. I think it will be a little clearer if we describe the behavior as either "splitting expressions" or "not splitting expressions". Thoughts?

Because I'm not familiar with the structure of the generated AST, I also found it useful to break out the part of the object that represents the list of expressions into a variable, to increase readability.

```javascript
const parseSheet = require('../lib/parse_sheet');

const expect = require('chai').expect;

describe('timelion parse_sheet function', function () {
  it(`doesn't split expressions on whitespace`, async function () {
    const data = ['.es() .es(404)'];
    const ast = parseSheet(data);

    const expressions = ast[0];
    expect(expressions.length).to.equal(1);
    expect(expressions[0].type).to.equal('chain');
  });]

  it('splits expressions on commas', function () {
    const data = ['.es(), .es(404)'];
    const ast = parseSheet(data);

    const expressions = ast[0];
    expect(expressions.length).to.equal(2);
    expect(expressions[0].type).to.equal('chain');
    expect(expressions[1].type).to.equal('chain');
  });

  it('splits expressions on newlines', function () {
    const data = [`.es()\n\r ,\n\r .es(404)`];
    const ast = parseSheet(data);

    const expressions = ast[0];
    expect(expressions.length).to.equal(2);
    expect(expressions[0].type).to.equal('chain');
    expect(expressions[1].type).to.equal('chain');
  });
});
```